front/lib/plans/stripe.ts

```diff
@@ -290,3 +290,81 @@ export async function cancelSubscriptionImmediately({
 
   return true;
 }
+
+const REPORT_USAGE_VALUES = ["MAU_1", "MAU_5", "MAU_10", "PER_SEAT"];
+export type StripeSubscriptionValidityType =
+  | { valid: true }
+  | { valid: false; message?: string };
+
+/**
+ * Checks that a subscription created in Stripe is usable by Dust
+ */
+export function checkStripeSubscriptionValid(
+  stripeSubscription: Stripe.Subscription
+): StripeSubscriptionValidityType {
```

Comment:
Why not using a `<Result>` type with appropriate errors to avoid introducing a new pattern?

Reply:
I understand the point. I initially did not do that because when we check validity usually we return a boolean, invalid is not necessarily an error (here it's boolean + message, since in addition we want the cause of invalidity). Analogy: it felt like writing `isEmailValid` that would throw instead of returning false when the email is not valid.

However, not strongly held... let's chat IRL, I could be convinced into renaming the function `errIfSubscriptionInvalid`

Reply:
as per IRL, turend into assert